Improve PHP comments conformity to Inline Documentation Standards

acicovic opened the pull request:
## Description
As described in #730, many of our comments do not conform to the [Inline Documentation Standards](https://developer.wordpress.org/coding-standards/inline-documentation-standards/). This PR is an attempt to quickly skim through the codebase and correct the most obvious comment issues.

Please note that:
- This is not a comment rewriting PR, where comments are rewritten to be better or provide more context.
- Changes to function comments that are in tests were avoided in order to avoid clashes with #837. Another PR will be made for tests.
- Minimal effort has been put in parts of the code that are deprecated or will be removed in version 4.0.0 of the plugin (Recommended Widget and some deprecated functions).

## Motivation and Context
- Solve as many comment issues as fast as possible.
- Related to #730.
- Related PRs for tests and JS will come in the future.

## How Has This Been Tested?
No code changes, automated tests pass.

acicovic committed:
Update PHP function comments to third person

See https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#language.

acicovic committed:
Remove return void from PHP function comments

See https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#1-functions-class-methods.

acicovic committed:
Attempt to limit some very long DocBlocks

See https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#line-wrapping,

acicovic committed:
Add missing full stops to PHP comments

pauarge reviewed (changes requested):
Do we want to remove `return void` from phpdoc? What does WP core do?

acicovic commented:
> Do we want to remove `return void` from phpdoc? What does WP core do?

See the last bullet point [here](https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#1-functions-class-methods). My understanding is that we should remove them (for void only). I think we should refer to this instead of inspecting core code, since the core itself might not be updated in certain parts.

pauarge commented:
I agree that we should look at the standards vs particular code. So yes, it looks like we should not use it. It's a bit counter intuitive to me, because if you are not putting it, it means that either you have forgot about it or it actually returns void. 

Having said that, I'd rather be consistent with the standards.

acicovic committed:
Update PHP file headers

See https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#6-file-headers.

acicovic committed:
PHP file headers: Remove author tag

acicovic committed:
PHP file headers: Remove copyright tag

See https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#other-tags.

acicovic committed:
PHP file headers: Remove license tag

License is still present in wp-parsely.php in the plugin header. See https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#other-tags.

acicovic committed:
PHP file headers: Remove package WordPress tag

See https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#other-tags.

acicovic committed:
PHP file headers: Update package tag to Parsely and Parsely\Tests

acicovic committed:
Merge branch 'develop' into update/improve-comments

acicovic committed:
Remove unused WP_ERROR

acicovic committed:
Merge branch 'update/improve-comments' of github.com:Parsely/wp-parsely into update/improve-comments

pauarge committed:
Merge branch 'develop' into update/improve-comments

pauarge commented on tests/Integration/Integrations/IntegrationsTest.php:
```diff
@@ -69,8 +69,6 @@ function( $integrations ) {
 // phpcs:disable Generic.Files.OneObjectStructurePerFile.MultipleFound
 /**
  * Class FakeIntegration2
- *
- * @package Parsely\Tests\Integrations
```
Maybe we shouldn't remove this one

acicovic replied:
Can you tell me what's your reasoning for this?

In the beginning I did start subcategorizing tests and it quickly became involved and too detailed (I think this is why I rolled back). If we categorize a type of test, my reasoning is that we should probably do it for all of them for consistency.

Let me know of what you think.

acicovic replied:
Just thinking out loud, but this could also change if we restructure our tests? So we may indeed want to look at it at that time?

pauarge replied:
That makes sense, we can leave it as is.

pauarge commented on wp-parsely.php:
```diff
@@ -1,11 +1,10 @@
  *
- * @package      Parsely\wp-parsely
- * @author       Parse.ly
- * @copyright    2012 Parse.ly
- * @license      GPL-2.0-or-later
+ * Requires any necessary files and loads the plugin.
+ *
+ * @package Parsely
```
Would this modifications have some effects on how WordPress.org represents our plugin?

acicovic replied:
Theoretically no, (although I suspect line 3 could be a potential problem) as WP looks at [header fields](https://developer.wordpress.org/plugins/plugin-basics/header-requirements/) and Packagist uses the [composer.json file](https://packagist.org/#how-to-submit-packages).

However, I've restored this just to be sure we're safe.

pauarge commented on src/Endpoints/class-related-api-proxy.php:
```diff
@@ -36,20 +36,19 @@ final class Related_API_Proxy {
 	 *
 	 * @param Parsely $parsely Instance of Parsely class.
-	 * @param Proxy   $proxy   Proxy object which does the actual calls to the Parse.ly API.
+	 * @param Proxy   $proxy   Proxy object which does the actual calls to the
+	 *                         Parse.ly API.
```
Is there any line character limitation that we're following for this line breaks?

acicovic replied:
[The standards](https://developer.wordpress.org/coding-standards/inline-documentation-standards/) state:

> DocBlock text should wrap to the next line after 80 characters of text. If the DocBlock itself is indented on the left 20 character positions, the wrap could occur at position 100, but should not extend beyond a total of 120 characters wide.

To conclude:

- I go for 80 chars anywhere possible.
- In some special cases I've gone for 100 and in one one case for 120, as the comments would take a lot of lines.
- In general, when we have long variable names and types (for example arrays), it's OK to go with 100 chars if this will save us an extra line.
- 120 chars is a last resort when having comments that will just take up too many lines.
- I also look at the code which is close to the comment. Having short lines of code with a really long comment doesn't look coherent.

Let me know if this answers the question.

pauarge replied:
That makes sense. Again, I personally don't like this rule (given we're in the age of ultra hdpi and wide displays), but if that's the standard, happy to adhere to it.

acicovic committed:
Restore wp-parsely.php file comments

acicovic committed:
Merge branch 'update/improve-comments' of github.com:Parsely/wp-parsely into update/improve-comments

GaryJones commented:
Nice work @acicovic!

acicovic commented:
> Nice work @acicovic!

Thank you! 😃
